meson.build

```diff
@@ -57,115 +77,132 @@ else
+      message('Detected macOS older than 10.15; will use the Boost implementation of filesystem')
+      boostmodules += ['filesystem'] # Boost::filesystem is used on systems where the standard filesystem is not available
+      conf.set('USE_BOOST_FS', 1)
     else
-      error('CppRestSdk not found but webserver build requested')
+      conf.set('USE_BOOST_FS', 0)
     endif
+  else
+    error('Can\'t determine macOS version.')
   endif
 else
-  cpprest=dependency('',required: false)
+  if not compiler.has_header('filesystem')
+    boostmodules += ['filesystem'] # Boost::filesystem is used on systems where the standard filesystem is not available
+    conf.set('USE_BOOST_FS', 1)
+  else
+    conf.set('USE_BOOST_FS', 0)
+  endif
 endif
 
-#Write configuration file
-configure_file(input : 'game/config.hh.in', output : 'config.hh', configuration : conf)
+# Write configuration file
+configure_file(input: 'game/config.hh.in', output: 'config.hh', configuration: conf)
 
 dependencies = [
-  dependency('pangocairo'),
-  dependency('sdl2'),
+  dependency('pangocairo', cmake_module_path: './cmake/Modules'),
+  dependency('sdl2', cmake_module_path: './cmake/Modules'),
   dependency('aubio'),
-  dependency('boost', modules : [ # Meson treats boost specially: no pkg-config supported
-    'filesystem',
-    'iostreams',
-    'locale',
-    'program_options',
-    'system'
-  ]),
-  dependency('epoxy'),
-  dependency('icu-uc'),
-  dependency('icu-i18n'),
-  dependency('icu-io'),
-  dependency('libpng'),
-  dependency('libjpeg'),
-  dependency('librsvg-2.0'),
-  dependency('libavcodec'),
-  dependency('libavformat'),
-  dependency('libavutil'),
-  dependency('libswscale'),
-  dependency('libswresample'),
-  dependency('portaudio-2.0'),
-  dependency('fontconfig'),
-  dependency('glm'),
-  dependency('libssl', required:get_option('usewebserver')), #Only needed with webserver
-  dependency('libcrypto', required:get_option('usewebserver')), #Idem
+  dependency('boost', modules: boostmodules),
+  dependency('epoxy', cmake_module_path: './cmake/Modules'),
+  dependency('icu-uc', cmake_module_path: './cmake/Modules'),
+  dependency('icu-i18n', cmake_module_path: './cmake/Modules'),
+  dependency('icu-io', cmake_module_path: './cmake/Modules'),
+  dependency('libpng', cmake_module_path: './cmake/Modules'),
+  dependency('libjpeg', cmake_module_path: './cmake/Modules'),
+  dependency('librsvg-2.0', cmake_module_path: './cmake/Modules'),
+  dependency('libavcodec', cmake_module_path: './cmake/Modules'),
+  dependency('libavformat', cmake_module_path: './cmake/Modules'),
+  dependency('libavutil', cmake_module_path: './cmake/Modules'),
+  dependency('libswscale', cmake_module_path: './cmake/Modules'),
+  dependency('libswresample', cmake_module_path: './cmake/Modules'),
+  dependency('portaudio-2.0', cmake_module_path: './cmake/Modules'),
+  dependency('fontconfig', cmake_module_path: './cmake/Modules'),
+  dependency('glm', cmake_module_path: './cmake/Modules'),
```

Comment:
Hmmm the cmake_module_path is in cmake a variable.  Not sure if meson can read or create variables and use it like that?

Reply:
It is the syntax meson uses to fallback to cmake modules to find its dependencies, after it fails from the default path (ie pkg-config, etc.)